salt/states/boto_iam.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+'''
+Manage IAM roles.
+=================
+
+.. versionadded:: TBD
+
+This module uses ``boto``, which can be installed via package, or pip.
+
+This module accepts explicit IAM credentials but can also utilize
+IAM roles assigned to the instance through Instance Profiles. Dynamic
+credentials are then automatically obtained from AWS API and no further
+configuration is necessary. More information available `here
+<http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
+
+It's also possible to specify ``key``, ``keyid`` and ``region`` via a profile, either
+passed in as a dict, or as a string to pull from pillars or minion config:
+
+.. code-block:: yaml
+
+    create-user:
+      boto_iam.user_present:
+        - name: myuser
+        - password: NewPassword$$1
+        - group: mygroup
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'fdkjsafkljsASSADFalkfjasdf'
+
+.. code-block:: yaml
+
+    creats-group:
+      boto_iam.group_present:
+        - name: mygroup
+        - users:
+          - myuser
+          - myuser1
+        - policy_name: test1
+        - policy: '{ "Version": "2012-10-17", "Statement": [ { "Effect": "Allow", "Action": "*", "Resource": "*" }]}'
```

Comment:
Hm. Is this just passing in a json string? If so, this is different from how the IAM role policies are defined, which is YAML, turned into json when sent to AWS. It's really nice for it to be YAML because it lets you treat the policy like other salt states, where you can use local variables, pillars, etc..